src/interactive-auth.js

```diff
@@ -235,13 +244,37 @@ InteractiveAuth.prototype = {
             throw new Error("submitAuthDict() called before attemptAuth()");
         }
 
+        if (!background && this._setBusyCallback) {
+            this._setBusyCallback(true);
+        }
+
+        // if we're currently trying a request, wait for it to finish
+        // as otherwise we can get multiple 200 responses which can mean
+        // things like multiple logins for register requests.
+        // (but discard any expections as we only care when its done,
+        // not whether it worked or not)
+        while (this._submitPromise) {
+            try {
+                await this._submitPromise;
+            } catch (e) {
+            }
+        }
+
         // use the sessionid from the last request.
         const auth = {
             session: this._data.session,
         };
         utils.extend(auth, authData);
 
-        await this._doRequest(auth, background);
+        try {
+            // NB. the 'background' flag is deprecated by the setBusy
```

Comment:
This deprecation should be copied up to the comments on the `background` param as well.